source/Halibut/Transport/SecureListener.cs

```diff
@@ -167,61 +171,73 @@ public int Start()
             var clientName = client.Client.RemoteEndPoint;
             var stream = client.GetStream();
 
-            using (var ssl = new SslStream(stream, true, AcceptAnySslCertificate))
+            using (cts.Token.Register(() =>
```

Comment:
This will allocate closure. Check how bad it is and eventually use the object state overload.